spec/models/group_spec.rb

```diff
@@ -125,6 +125,22 @@
     context 'with bob in the team' do
       before { team.people << bob }
 
+      context 'and bob is a team leader' do
+        before do
+          @leadership = bob.memberships.first
+          @leadership.leader = true
+          @leadership.save!
+        end
```

Comment:
Could you convert this to a `let` block? I'm not sure if the membership can be created here explicitly, but if the existing has to be changed, it could still be something like:

```ruby
let(:leadership) { bob.memberships.first.update(leader: true) }
```

And the use it in the bellow assertions.